### ISSUE TITLE
MD033 allowed_elements highlights the wrong element in mixed element scenarios

### ISSUE THREAD

**Issue by serotta58** [DavidAnson/markdownlint#179]
In VS Code, the markdownlint allowed_elements setting works for other html elements that I have tried, but not `<details>`.  For example, with the following there are no warnings for h1 or p (as expected), but details still gets a warning :
```
"markdownlint.config": {
    "MD013": false,
    "no-inline-html": {
        "allowed_elements": [ "details", "h1", "p" ]
    }
}
```

**Comment by DavidAnson** [comment]
Creating an empty directory with the following Markdown file in it, opening the folder in VS Code with the `markdownlint` extension, and viewing the file reports `MD033` violations for both instances of `p` and `details` as expected:

`179.md`

```markdown
# 179

Text

<p>html</p>

Text <p>html</p> text

Text

<details>html</details>

Text <details>html</details> text
```

Adding the following configuration file to that directory suppresses those violations as expected:

`.markdownlint.json`

```json
{
  "no-inline-html": {
    "allowed_elements": [ "details", "p" ]
  }
}
```

Could you please provide an example file/scenario where this does not work for you?

**Comment by serotta58** [comment]
'Hi David,

The problem seems to be with <details> with an embedded <summary> tag.  If
you past the following line into your code it will incorrectly warn about
the <details>:

<details><summary>Short version</summary>Longer version with more detail</
details>

Except for this, the settings work in a local `.markdownlint.config` file,
but NOT in the VSCode user settings file.  Mine is
at /Users/Glen/Library/Application Support/Code/User/settings.json and is
attached.  The markdownlint settings have no effect.

**Comment by serotta58** [comment]
Update:  When I add "summary" to the allowed_elements array, then the <details> below is no longer marked with a warning.  The fact that `<details>` was marked and not `<summary>` threw me.  Shouldn't the summary be marked if that is the real problem?

`<details><summary>Short version</summary>Longer version with more detail</details>`

**Comment by DavidAnson** [comment]
Thanks! It sounds like the rule is working correctly, but the highlighting is targeting the wrong element. I will look into that.

Regarding configuration being overwritten, I’m not sure if you are having an issue but please see the following for more information about how that works: https://github.com/DavidAnson/vscode-markdownlint/issues/70

**Comment by serotta58** [comment]
Thanks David.  The other issue was due to the local configuration file
overriding the VSCode user settings, and was fixed when I removed the local
config file.  I was assuming the effects were cumulative, or would add to
each other, but now I understand that there is a hierarchy and only one set
of settings wins.  That was not clear from the sections of the
documentation that I had read.  But I do see the logic behind it.

**Comment by tdalis** [comment]
I don't know if this is the right thread, but I get an issue with MD013 and when I try to allow elements, I get error "Property allowed_elements is not allowed" - I am using VSC version: 1.99.3 on a Mac
